apis/python/src/tiledb/vector_search/ivf_flat_index.py

```diff
@@ -295,8 +302,41 @@ def query_internal(
                 num_partitions=num_partitions,
                 num_workers=num_workers,
                 config=self.config,
+                distance_metric=self.distance_metric,
             )
 
+    def update(self, vector: np.array, external_id: np.uint64, timestamp: int = None):
+        if self.distance_metric == vspy.DistanceMetric.COSINE:
+            vector = normalize_vector(vector)
+        super().update(vector, external_id, timestamp)
+
+    def update_batch(
+        self, vectors: np.ndarray, external_ids: np.array, timestamp: int = None
+    ):
+        if self.distance_metric == vspy.DistanceMetric.COSINE:
+            vectors = normalize_vectors(vectors)
+        super().update_batch(vectors, external_ids, timestamp)
+
+    def query(
+        self,
+        queries: np.ndarray,
+        k: int,
+        driver_mode: Mode = None,
+        driver_resources: Optional[str] = None,
+        driver_access_credentials_name: Optional[str] = None,
+        **kwargs,
+    ):
+        if self.distance_metric == vspy.DistanceMetric.COSINE:
+            queries = normalize_vectors(queries)
+        return super().query(
+            queries,
+            k,
+            driver_mode,
+            driver_resources,
+            driver_access_credentials_name,
+            **kwargs,
+        )
```

Comment:
(Minor) This seems okay, but we could also just update these methods in `index.py` to have:
```
if self.index_type == "IVF_FLAT" and self.distance_metric == vspy.DistanceMetric.COSINE:
    vector = normalize_vector(vector)
```
Will be less code and make things easier to reason about.

Reply:
I am down to do it either way, just seemed like a clean way to do it since we are already using polymorphism

Reply:
Sure, sounds good, your choice, I'm fine with either approach